utils/utils.go

```diff
@@ -173,3 +179,33 @@ func GetDefaultProjectId() (string, error) {
 func getServiceAccountJson() string {
 	return viper.GetString("google.account")
 }
+
+// ExtractDefaultLabels creates a map[string]string of labels that should be
+// applied to a resource on creation if the resource supports labels.
+// These include the organization, space, and instance id.
+func ExtractDefaultLabels(instanceId string, details brokerapi.ProvisionDetails) map[string]string {
+	labels := map[string]string{
+		"pcf-organization-guid": details.OrganizationGUID,
+		"pcf-space-guid":        details.SpaceGUID,
+		"pcf-instance-id":       instanceId,
+	}
+
+	// After v 2.14 of the OSB the top-level organization_guid and space_guid are
+	// deprecated in favor of context, so we'll override those.
+	requestContext := map[string]string{}
+	json.Unmarshal(details.GetRawContext(), &requestContext) // explicitly ignore parse errors
+	if orgGuid, ok := requestContext["organization_guid"]; ok {
+		labels["pcf-organization-guid"] = orgGuid
+	}
+
+	if spaceGuid, ok := requestContext["space_guid"]; ok {
+		labels["pcf-space-guid"] = spaceGuid
+	}
+
+	sanitized := map[string]string{}
+	for key, value := range labels {
+		sanitized[key] = invalidLabelChars.ReplaceAllString(value, "_")
```

Comment:
Would it be a good idea to log when a key needed to be sanitized?

Reply:
I added some documentation about billing to include this now.